src/bindings/c/include/openvino/ov_common.h

```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a common header file for the C API
+ *
+ * @file ov_common.h
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#ifdef __cplusplus
+#    define OPENVINO_C_API_EXTERN extern "C"
+#else
+#    define OPENVINO_C_API_EXTERN
+#endif
+
+#if defined(OPENVINO_STATIC_LIBRARY) || defined(__GNUC__) && (__GNUC__ < 4)
+#    define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __VA_ARGS__
+#    define OV_NODISCARD
+#else
+#    if defined(_WIN32)
+#        define OPENVINO_C_API_CALLBACK __cdecl
+#        ifdef openvino_c_EXPORTS
+#            define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __declspec(dllexport) __VA_ARGS__ __cdecl
+#        else
+#            define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __declspec(dllimport) __VA_ARGS__ __cdecl
+#        endif
+#        define OV_NODISCARD
+#    else
+#        define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __attribute__((visibility("default"))) __VA_ARGS__
+#        define OV_NODISCARD        __attribute__((warn_unused_result))
+#    endif
+#endif
+
+#ifndef OPENVINO_C_API_CALLBACK
+#    define OPENVINO_C_API_CALLBACK
+#endif
+
+#define MAX_DIMENSION 8
+
+typedef struct ov_core ov_core_t;
+typedef struct ov_output_const_node ov_output_const_node_t;
+typedef struct ov_output_node ov_output_node_t;
+typedef struct ov_model ov_model_t;
+typedef struct ov_compiled_model ov_compiled_model_t;
+typedef struct ov_infer_request ov_infer_request_t;
+
+typedef struct ov_property ov_property_t;
+typedef void* ov_property_value_t;
```

Comment:
Can you give some sample on how to implement ov::Any in C code? I need some ideas about what you think.

Reply:
Maybe something like that:
```
struct any {
void* data;
size_t size;
typeid type;
```
`typeid` can be implemented in different manner, but the main purpose is to provide an information about data type.
any in C can wrap only known C types.

Reply:
typeid is not supported in C code, to support it we need to implement additional function, and also need pay more attention on memory leak of void * data, it will bring lots of potential trouble to user.  It is not a good idea, I don't like such idea! 
I still think the union structure is better than yours.

Reply:
All right, though I don't like this idea, but I still implement them, please review.

Reply:
done